java/arcs/android/sdk/host/IntentArcHostAdapter.kt

```diff
@@ -131,6 +131,6 @@ class IntentArcHostAdapter(
          * The maximum amount of time to wait for an [ArcHost] to process an [Intent]-base RPC call.
          * This timeout ensures requests don't wait forever.
          */
-        const val ARCHOST_INTENT_TIMEOUT = 2000L
+        const val ARCHOST_INTENT_TIMEOUT = 5000L
```

Comment:
This worries me a bit ... 2s is already a pretty long timeout (note I have no context as to what these calls are doing).

Reply:
I didn't dig into this a whole lot.

I saw this timeout trigger if I ran a bunch of tests concurrently, or every once in a while on GCB.

The tests that fail involve Robolectric, so my assumption was that Robolectric was having trouble keeping up.

I would like to make sure @cromwellian has a chance to weigh in before submitting this.

Reply:
The way Allocator communicates with ArcHost is by sending an Intent.   It then waits for a ResultReceiver before resuming.  This waits a maximum of (now 5) seconds to start an arc. This is so that Allocator.startArcForPlan() appears as a synchronous API and doens't return until all ArcHosts are started.

Within Robolectric, it shouldn't take anywhere near this time, as the Android SDK is short-circuited and the onStartCommand() of the Service is invoked synchronously. 

However, while this works flawlessly locally, for some reason on GCB, it's flaky.  Eventually, this logic probably won't wait. Instead, Allocator.startArcForPlan() will return an Arc object instead of an ID, and you can add listeners to check for asynchronous status (all allhosts started, or there was an error)

Reply:
Thanks for the explanation, Ray. Seems like a reasonable workaround.